examples/automm/distillation/automm_distillation.py

```diff
@@ -0,0 +1,145 @@
+        },
+        time_limit=args.time_limit,
+        seed=args.seed,
+    )
+    nodistill_result = nodistill_predictor.evaluate(data=valid_df, metrics=GLUE_METRICS[glue_task]["eval"])
+
+    ### Distill and evaluate a student model
+    from autogluon.multimodal.constants import MODEL, DATA, OPTIMIZATION, ENVIRONMENT, DISTILLER
+    config = {
+        MODEL: f"fusion_mlp_image_text_tabular",
+        DATA: "default",
+        DISTILLER: "default",
+        OPTIMIZATION: "adamw",
+        ENVIRONMENT: "default",
+    }
+    student_predictor = AutoMMPredictor(label="label", eval_metric=GLUE_METRICS[glue_task]["val"])
+    student_predictor.fit(
+        train_df,
+        config=config,
+        hyperparameters={
+            "env.num_gpus": args.num_gpu,
+            "optimization.max_epochs": args.max_epochs,
+            "model.hf_text.checkpoint_name": args.student_model,
+            #"optimization.learning_rate": 1.0e-4,
+            #"optimization.weight_decay": 1.0e-3,
+            #"distiller.temperature": args.temperature,
+            #"distiller.hard_label_weight": args.hard_label_weight,
+            #"distiller.soft_label_weight": args.soft_label_weight,
+        },
+        teacher_predictor=teacher_predictor,
+        time_limit=args.time_limit,
+        seed=args.seed,
+    )
+    start = time()
+    student_result = student_predictor.evaluate(data=valid_df, metrics=GLUE_METRICS[glue_task]["eval"])
+    student_usedtime = time() - start
+
+    ### Print distillation's performance
+    print("Distillation Result:")
+    print("Teacher Model: %s" % args.teacher_model)
+    print("Student Model: %s" % args.student_model)
+    for k in teacher_result:
+        print(f"For metric {k}:")
+        print("Teacher Model's %s: %.6f" % (k, teacher_result[k]))
+        print("Pretrained Model's %s: %.6f" % (k, nodistill_result[k]))
+        print("Student Model's %s: %.6f" % (k, student_result[k]))
+        print(
+            "Distillation Ratio (the fraction of the teacher's performance achieved by the student): %.6f"
+            % (
+                float(student_result[k] - nodistill_result[k])
+                / float(teacher_result[k] - nodistill_result[k])
+            )
+        )
+    print("Teacher Model's time: %.6f" % teacher_usedtime)
+    print("Student Model's time: %.6f" % student_usedtime)
+    print("speed up: %.6fx" % (teacher_usedtime / student_usedtime))
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--glue_task", default="qnli", type=str)
+    parser.add_argument("--teacher_model", default="google/bert_uncased_L-12_H-768_A-12", type=str)
+    parser.add_argument("--student_model", default="google/bert_uncased_L-4_H-768_A-12", type=str)
+    parser.add_argument("--seed", default=123, type=int)
+    parser.add_argument("--max_epochs", default=1000, type=int)
+    parser.add_argument("--time_limit", default=7200, type=int)
```

Comment:
We can also use `None` as the default time-limit so these models will be trained until convergence.